packages/svelte/src/internal/client/proxy.js

```diff
@@ -172,7 +172,11 @@ export function proxy(value, options, parent = null, prev) {
 
 			v = Reflect.get(target, prop, receiver);
 
-			if (is_proxied_array && array_methods.includes(/** @type {string} */ (prop))) {
+			if (
+				is_proxied_array &&
+				array_methods.includes(/** @type {string} */ (prop)) &&
+				options?.onchange != null
```

Comment:
this should be before the includes call

Reply:
Yeah i already moved it, i realised as soon as i committed 😄